include/kernel/vm.h

```diff
@@ -179,6 +179,10 @@ size_t pmm_free_kpages(void *ptr, uint count);
 /* physical to virtual */
 void *paddr_to_kvaddr(paddr_t pa);
 
+/* a hint as to which virtual addresses will be returned by pmm_alloc_kpages */
+void *first_mapping_address(void);
```

Comment:
Would rather name it to be semi-consistent something like kvaddr_get_range() that returns a void *, size_t. The implementation detail of how it works should be hidden by the implementation and will probably be made smarter at some point.